### ISSUE TITLE
So to connect to someone else your gota dox yourself?

### ISSUE THREAD

**Issue by Sonickirbystar** [Cracko298/MC3DS-Multiplayer-Mod#3]
"Requires The Clients' Public IP Address (IPv4 or IPv6), and a valid Port Number."

**Comment by Cracko298** [comment]
Alot of games and fan creations do this.
You'll need to know the IP Address, and Port Number of the Person you're trying to play online with.

Examples include: Sonic RB2 Kart, The ART of War, and yes even this Software.

**Comment by Sonickirbystar** [comment]
Well I guess since you know their IP and they know yours its equal = equal.

**Comment by Cracko298** [comment]
> Well I guess since you know their IP and they know yours its equal = equal.

This project isn't aimed towards doxxing other people.
Instead creating a fun and unique gameplay experience.

Once i have the final build running all data should be encrypted and decrypted.

**Comment by Sonickirbystar** [comment]
That sounds cool but another question is what happens if you play with someone who has their game modded like killing a pig gives you 64 diamonds or creepers are rideable connects to someone or has someone connecting to them with a unmodified (expect the online plugin your making) game?

**Comment by Cracko298** [comment]
No mod support will be added.
Only client-side mods will work, such as gfx, sounds, font, shades, textured, etc.

The plugin itself is entirely locked down and requires the source code directly to change it.

**Comment by Sonickirbystar** [comment]
Wait so if its locked down then would it be possible to make a IP Decoder? Like it decodes your IP to a string of numbers like a friend code sorta. So you can give the decoded IP to the person and they give you their decoded IP to use which gets decoded by your plugin? Obviously its at risk of being decoded but its less idk doxy?

**Comment by Cracko298** [comment]
No, no decoder will be able to get any data sent to and from the IP address.
Instead, it'll be encrypted with XOR or AES256 depending on which is easier to implement.

**Comment by trevorwatkins1129** [comment]
> Wait so if its locked down then would it be possible to make a IP Decoder? Like it decodes your IP to a string of numbers like a friend code sorta. So you can give the decoded IP to the person and they give you their decoded IP to use which gets decoded by your plugin? Obviously its at risk of being decoded but its less idk doxy?

I think what they mean is instead of having the join thing be like "124.87.678.501" it would be like "123-456-789" or something and get decoded into the ip.

**Comment by Cracko298** [comment]
> > Wait so if its locked down then would it be possible to make a IP Decoder? Like it decodes your IP to a string of numbers like a friend code sorta. So you can give the decoded IP to the person and they give you their decoded IP to use which gets decoded by your plugin? Obviously its at risk of being decoded but its less idk doxy?
> 
> I think what they mean is instead of having the join thing be like "124.87.678.501" it would be like "123-456-789" or something and get decoded into the ip.


It's more of a Special Key is used from a Database somewhere. I still haven't wrote the code to do it, but an example would encrypt the IP Address rather than encode.

So the first IP you provided using an

Example Key: a373df99818f76466d082616e500b472

Will Create The Following Data:
e305d54c98690c3525ff3e7ff2e60b37


Of course, this key will NOT be public and will always be random between two devices.
